### DISCUSSION TITLE
실수를 삭제할때 태그와의 연관관계까지 전부 삭제하지 고민이에요!

### DISCUSSION THREAD

**Discussion by woodowoon** [coding-union-kr/siltarae-be#156] [General]
### 현재 동작 방식
현재는 실수를 삭제한다면 실수의 deletedAt 만 현재 시간으로 바뀌는 식으로 개발이 되어있습니다.

### 변경하고 싶은 방식
실수를 삭제할때 태그와의 연관관계까지 끊어줘야할까? 라는 생각을 하고 있습니다

### 왜 변경하고 싶은지?
태그를 조회할때 태그에 속해있는 실수 개수를 가져오는데 getMistakes.size 를 하게되면 삭제된 실수의 개수까지 카운팅 되는 현상이 발생하고있어요. 이 부분을 수정하려고 하면서 생각을 해보니깐 실수가 삭제가 되었으므로 태그와의 연관관계도 끊어줘야하지 않을까? 라는 생각까지 도달하게 되었어요.

### 변경했을 때 장점
- 태그를 조회할때 태그에 속해있는 실수 개수를 가져올때 별다른 코드 수정 없이 가져와도 괜찮아요! 

### 변경했을 때 단점
- 추후에 실수 살리기 등의 추가 기능이 발생하게 된다면 이미 태그와 실수와의 연관관계가 끊어져서 해당 실수가 어떤 태그와 연결되어있었는지 알 수 가 없습니다.

**Comment by seovee** [comment]
이부분 한번 화면공유해서 보여주실 수 있나여?
어떤 내용인지 이해는 되지만, 한번 봐야할거 같습니다.

**Comment by gorapaduckoo** [comment]
저는 연관관계도 끊어주는게 좋을것같다고 생각해요. 매번 해당 실수가 삭제되어있는지까지 조회하면 태그 조회 한번에 비용이 너무 커질것같아요. 현재 연관관계 테이블은 실수id - 태그id만 저장하는 다대다 상태니까 테이블 설계를 바꾸지 않는 이상은 soft delete는 어렵고 hard delete만 가능하겠네요🤔

**Comment by woodowoon** [comment]
그럼 2가지 방법이 있을 것 같아요
1. 실수 자체를 hard delete 하는 것과 동시에 태그와의 연관관계까지 끊어주는 방법
2. 현재 실수는 soft delete 하고있는데 soft delete 하는 것을 그대로 두고 태그와의 연관관계만 끊어줄 것

저는 저희가 테이블 설계나 지금까지 모든 데이터들을 soft delete 하는 방향으로 설계했었기때문에 2번의 방법이 조금 더 나은 것 같은데 어떻게 생각하시나요?

**Reply by gorapaduckoo** [reply]
제 의견으로는 실무에서 hard delete를 잘 사용하지 않고, 실수는 연관관계도 복잡해서 2번이 더 나을것같아요. 테이블 설계를 바꾸면 삭제할 때는 편하겠지만, 사이드 이펙트도 크고 설정한 태그를 변경할 때 번거로울 것 같아서 2번이 현실적으로 가장 무난한 선택지일 것 같습니다!

**Reply by woodowoon** [reply]
좋아요 그렇다면 2번의 방법으로 이슈 올리겠습니다!
